pkg/controller/plan/adapter/ova/builder.go

```diff
@@ -0,0 +1,594 @@
+	TemplateWorkloadLabel = "workload.template.kubevirt.io/server"
+	TemplateFlavorLabel   = "flavor.template.kubevirt.io/medium"
+)
+
+// Operating Systems
+const (
+	DefaultWindows = "win10"
+	DefaultLinux   = "rhel8.1"
+	Unknown        = "unknown"
+)
+
+// Annotations
+const (
+	// CDI import backing file annotation on PVC
+	AnnImportBackingFile = "cdi.kubevirt.io/storage.import.backingFile"
+)
+
+// Regex which matches the snapshot identifier suffix of a
+// OVA disk backing file.
+var backingFilePattern = regexp.MustCompile("-\\d\\d\\d\\d\\d\\d.vmdk")
+
+// vSphere builder.
+type Builder struct {
+	*plancontext.Context
+	// MAC addresses already in use on the destination cluster. k=mac, v=vmName
+	macConflictsMap map[string]string
+}
+
+// Get list of destination VMs with mac addresses that would
+// conflict with this VM, if any exist.
+func (r *Builder) macConflicts(vm *model.VM) (conflictingVMs []string, err error) {
+	if r.macConflictsMap == nil {
+		list := []ocp.VM{}
+		err = r.Destination.Inventory.List(&list, base.Param{
+			Key:   base.DetailParam,
+			Value: "all",
+		})
+		if err != nil {
+			return
+		}
+
+		r.macConflictsMap = make(map[string]string)
+		for _, kVM := range list {
+			for _, iface := range kVM.Object.Spec.Template.Spec.Domain.Devices.Interfaces {
+				r.macConflictsMap[iface.MacAddress] = path.Join(kVM.Namespace, kVM.Name)
+			}
+		}
+	}
+
+	for _, nic := range vm.NICs {
+		if conflictingVm, found := r.macConflictsMap[nic.MAC]; found {
+			for i := range conflictingVMs {
+				// ignore duplicates
+				if conflictingVMs[i] == conflictingVm {
+					continue
+				}
+			}
+			conflictingVMs = append(conflictingVMs, conflictingVm)
+		}
+	}
+
+	return
+}
+
+// Create DataVolume certificate configmap.
+// No-op for OVA.
+func (r *Builder) ConfigMap(_ ref.Ref, _ *core.Secret, _ *core.ConfigMap) (err error) {
+	return
+}
+
+func (r *Builder) PodEnvironment(vmRef ref.Ref, sourceSecret *core.Secret) (env []core.EnvVar, err error) {
+	vm := &model.VM{}
+	err = r.Source.Inventory.Find(vm, vmRef)
+	if err != nil {
+		err = liberr.Wrap(
+			err,
+			"VM lookup failed.",
+			"vm",
+			vmRef.String())
+		return
+	}
+
+	env = append(
+		env,
+		core.EnvVar{
+			Name:  "V2V_vmName",
+			Value: vm.Name,
+		},
+	)
+
+	if strings.HasSuffix(vm.OvaPath, ".ova") {
+		env = append(env,
+			core.EnvVar{
+				Name:  "V2V_diskPath",
+				Value: vm.OvaPath,
+			},
+			core.EnvVar{
+				Name:  "V2V_provider",
+				Value: "ova",
+			})
+	} else {
+		for _, disk := range vm.Disks {
+			env = append(env,
+				core.EnvVar{
+					Name:  "V2V_diskPath",
+					Value: getDiskSourcePath(vm.OvaPath, disk.Name),
+				})
+		}
+		env = append(env,
+			core.EnvVar{
+				Name:  "V2V_provider",
+				Value: "ovf",
+			})
+	}
+	return
+}
+
+// Build the DataVolume credential secret.
+func (r *Builder) Secret(vmRef ref.Ref, in, object *core.Secret) (err error) {
+	// TODO only if we want to save some data for DV.
```

Comment:
what does it mean? if we need an empty volume to store information on, why do we need a secret?

Reply:
we don't. this comment should be removed.